src/api/osaamiset.ts

```diff
@@ -0,0 +1,86 @@
+import { client } from './client';
+import { components } from './schema';
+
+type OsaaminenDto = components['schemas']['OsaaminenDto'];
+const cache = new Map<string, OsaaminenDto>();
+const maxQueryLength = 4096;
+const queryParamOverhead = 5; // uri=...&
+
+export const osaamiset = {
+  /**
+   * Combines objects referencing osaamiset with osaamiset.
+   *
+   * @param objects Collection of objects to combine with osaamiset.
+   * @param key Property accessor for the osaaminen uri.
+   * @param combiner Function to combine the object with the osaaminen.
+   * @param signal Abort signal.
+   * @returns An array of objects combined with osaamiset.
+   */
+  combine: async <T, R>(
+    objects: Iterable<T> | undefined,
+    key: (o: T) => string,
+    combiner: (object: T, osaaminen: OsaaminenDto) => R,
+    signal?: AbortSignal,
+  ) => {
+    if (!objects) {
+      return [];
+    }
+    const chunks: string[][] = [];
```

Comment:
This previous comment would be worth keeping to explain the motivation for an unfamiliar reader (unless already completely covered by the method-level comment)

`// the list of competences can be quite long, so we split the request into chunks to avoid exceeding the maximum URL length`